File: gluecodium/src/test/resources/smoke/visibility/output/android/com/example/smoke/InternalClass.java
/*
 *

 */

package com.example.smoke;

import com.example.NativeBase;

class InternalClass extends NativeBase {

    protected InternalClass(final long nativeHandle) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }

    private static native void disposeNativeHandle(long nativeHandle);
}

File: gluecodium/src/test/resources/smoke/visibility_attribute/output/android-kotlin/com/example/smoke/InternalClass.kt
/*

 *
 */

@file:JvmName("InternalClassExtensions")


package com.example.smoke

import com.example.NativeBase

internal class InternalClass : NativeBase {



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}





    @JvmSynthetic @JvmName("fooBar") internal external fun fooBar() : Unit




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}

